scripts/PrepareKubernetesTarget.ps1

```diff
@@ -0,0 +1,39 @@
+﻿$rnd = Get-Random -Minimum 1000000 -Maximum 9999999
+$svcName = "sqltoolsvc" + $rnd
+$pwd = "Yukon" + $rnd
+$env:KUBECONFIG = $env:AGENT_WORKFOLDER + "\kubeconfig.centralus.json"
+
+$kube = $env:AGENT_WORKFOLDER + "\kubectl.exe"
+
+iex "$kube run $svcName --image=sqltoolscontainers.azurecr.io/sql2017linux --port=1433 --env ACCEPT_EULA=Y --env SA_PASSWORD=$pwd"
+iex "$kube expose deployment $svcName --type=LoadBalancer"
+
+do
+{
+    $svc = iex "$kube describe service $svcName"
+    Write-Host "Service Configuration: $svc"
+    $endpoint = $svc -match "Endpoints:(.+):1433"
+    Write-Host "Endpoint: $endpoint"
+# the output of kubectl is an object array
+} while ($endpoint -eq $null -or $endpoint.Length -eq 0)
+
+$endpoint = $endpoint.Split(":")[1].Trim()
+Write-Host "##vso[task.setvariable variable=k8EndPoint;]"$endpoint
+Write-Host "##vso[task.setvariable variable=k8ServiceName;]"$svcName
+
+$settingsOutput = "{ `"mssql.connections`":[ "
```

Comment:
does powershell not have a json serializer such that you don't need to hand build it? Not that this is hard to read